srcdeps-core/src/main/java/org/srcdeps/core/Gavtc.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2015-2017 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.srcdeps.core;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.StringTokenizer;
+
+/**
+ * An immutable {@link #groupId}, {@link #artifactId}, {@link #version}, {@code type}, {@code classifier} tuple. Note
+ * that only {@code classifier} can be {@code null}.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class Gavtc extends Gav {
+
+    /**
+     * Returns a new {@link Gavtc} instance parsed out of the given {@code gavtcString}.
+     *
+     * @param gavtcString
+     *            the string to parse, something of the form {@code groupId:artifactId:version:type:classifier}
+     * @return a new {@link Gavtc} instance parsed out of the given {@code gavtcString}
+     */
+    public static Gavtc of(String gavtcString) {
+        StringTokenizer st = new StringTokenizer(gavtcString, ":");
```

Comment:
Couldn't you split it in two parts, the first calling the `super` version of `of(String)` and the second parsing only the `tc`?

Reply:
Would it not mean creating one StringTokenizer instance and one intermediate `Gav` instance that would have to be thrown away? Sounds ineffective.

Reply:
Indeed, when you put it this way :)